libafl/src/executors/inprocess/mod.rs

```diff
@@ -90,16 +90,24 @@ where
         input: &I,
     ) -> Result<ExitKind, Error> {
         *state.executions_mut() += 1;
+
+        // unsafe operations for settings up pointers
         unsafe {
             let executor_ptr = ptr::from_ref(self) as *const c_void;
             self.inner
                 .enter_target(fuzzer, state, mgr, input, executor_ptr);
+            // the first hook is special
+            // it is for setting crash handlers
+            self.inner.hooks.0.enter_target_hooks();
```

Comment:
At this point, why store this dude in the hooks at all?

Reply:
because only this hook has the pointer to the crash handler.
but you are right. the crash handler's address won't move. there's no reason to write this pointer at all.